src/main/java/org/apache/mesos/scheduler/plan/Block.java

```diff
@@ -63,38 +32,14 @@ static Status getStatus(Block block) {
      * could be found. This is only called if {@link #start()} returned a non-{@code null}
      * {@link OfferRequirement}.
      */
-    void updateOfferStatus(Collection<Protos.Offer.Operation> operations);
-
-    /**
-     * Forcefully restarts the block, putting it into a PENDING state, waiting to be resumed with
-     * a call to {@link start()}.
-     */
-    void restart();
+    void updateOfferStatus(Collection<Offer.Operation> operations);
 
     /**
-     * Forcefully marks the block as COMPLETE, cancelling any work that hasn't started or that's
-     * currently in progress.
+     * Thrown on invalid Block construction attempt.
      */
-    void forceComplete();
-
-    /**
-     * Provides the Block with a recent {@link Protos.TaskStatus} update which was received from
-     * Mesos.
-     */
-    void update(Protos.TaskStatus status);
-
-    /**
-     * Returns the unique identifier of the block.
-     */
-    UUID getId();
-
-    /**
-     * Returns a user-visible message describing the current status of the Block.
-     */
-    String getMessage();
-
-    /**
-     * Returns a user-visible name describing the purpose of the Block.
-     */
-    String getName();
+    class InvalidException extends Exception {
```

Comment:
API Style nit: Can we also rename this to something more explicit ? Maybe `InvalidElementException` or `InvalidBlockException` ?